### PR TITLE
Dbeaver/dbeaver#23135 cannot create driver instance

### PR DESCRIPTION
**Pull request by serjiokov**

### CODE REVIEW

**Comment by serjiokov** [comment]
1- positive case 
image
2 -negative (drivers doesn't exist)
image

**Comment by serjiokov** [comment]
3 - cancelation
image

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import org.eclipse.core.runtime.jobs.JobChangeAdapter;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java]: Redundant changes in this class

File: plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionWizard.java [hunk 2]
```diff
@@ -37,6 +37,7 @@
 import org.jkiss.dbeaver.model.connection.DBPDriver;
 import org.jkiss.dbeaver.model.connection.DBPDriverSubstitutionDescriptor;
 import org.jkiss.dbeaver.model.connection.DBPNativeClientLocation;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionWizard.java]: Also redundant changes

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/exec/DBCMessages.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jkiss.dbeaver.model.exec;
+
+import org.eclipse.osgi.util.NLS;
+
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/exec/DBCMessages.java]: Let's move all messages into ModelMessages

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/jobs/ConnectJob.java [hunk 4]
```diff
@@ -86,9 +86,13 @@ protected IStatus run(DBRProgressMonitor monitor)
         }
         catch (Throwable ex) {
             connectError = ex;
-            connectStatus = GeneralUtils.makeExceptionStatus(ex);
+            if (monitor.isCanceled()) {
+                connectStatus = GeneralUtils.makeExceptionStatus(IStatus.CANCEL, ex);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/runtime/jobs/ConnectJob.java]: You can use Status.CANCEL_STATUS instead

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java [hunk 5]
```diff
@@ -569,6 +570,13 @@ private static IStatus makeExceptionStatus(int severity, Throwable ex, boolean n
         if (ex instanceof CoreException) {
             return ((CoreException) ex).getStatus();
         }
+        if (ex instanceof DBCDriverFilesMissingException) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java]: Why extra logic here? This util method shouldn't be aware about driver errors, it is too generic

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/DriverEditDialog.java [hunk 6]
```diff
@@ -1020,16 +1054,25 @@ protected void createButtonsForButtonBar(Composite parent) {
         @Override
         protected void buttonPressed(int id) {
             if (id == IDialogConstants.RETRY_ID) {
-                UIUtils.asyncExec(() -> {
-                    DriverEditDialog dialog = new DriverEditDialog(
-                        UIUtils.getActiveWorkbenchShell(),
-                        dataSource.getContainer().getDriver());
-                    dialog.open();
-                });
+                if (dataSource != null) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/dialogs/driver/DriverEditDialog.java]: Why do we need these changes?

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by serjiokov** [comment]
Closed as nonactual

**Pull request closed without merging**